fdbclient/FDBTypes.h

```diff
@@ -665,6 +665,10 @@ struct RangeResultRef : VectorRef<KeyValueRef> {
 		serializer(ar, ((VectorRef<KeyValueRef>&)*this), more, readThrough, readToBegin, readThroughEnd);
 	}
 
+	int logicalSize() const {
+		return VectorRef<KeyValueRef>::expectedSize() - VectorRef<KeyValueRef>::size() * sizeof(KeyValueRef);
```

Comment:
The VectorRef<T>::expectedSize() doesn't include the KeyValueRef data structure's overhead. 
It's implementation is https://github.com/apple/foundationdb/blob/092af661985d871c6a68cf3f38e379859339b705/flow/Arena.h#L1099-L1106 

KeyValueRef is https://github.com/apple/foundationdb/blob/092af661985d871c6a68cf3f38e379859339b705/fdbclient/FDBTypes.h#L396

Reply:
Is it [initialized](https://github.com/apple/foundationdb/blob/092af661985d871c6a68cf3f38e379859339b705/flow/Arena.h#L1102) as the sizeof(KeyValueRef) * size?